File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, l: List[int], k: int) -> int:
        c = 0
        m = 0
        d = {}
        for i in range(0,len(l)):
            m += l[i]
            if i==0 and m-k==0:
                c+=1
            elif m==k:
                c+=1
            if (m-k) in d:
                a = d[m-k]
                c+= len(a)
            if m not in d:
                y = []
                y.append(i)
                g = tuple((y))
                d[m]=g
            else:
                x = list(d[m])
                x.append(i)
                z = tuple((x))
                d[m] = z
        return c

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int,int>map;
        long long m = 0;
        int c = 0;
        for(int i = 0; i < nums.size(); i++){
            m += nums[i];
            if(i==0 and m==k) c++;
            else if(m==k) c++;
            if(map.find(m-k)!=map.end()){
                c+=map[m-k];
            }
            map[m]++;;
          
        }
        return c;
    }
};